esphome/components/pmsa003i/pmsa003i.cpp

```diff
@@ -75,35 +87,48 @@ void PMSA003IComponent::update() {
 }
 
 bool PMSA003IComponent::read_data_(PM25AQIData *data) {
-  const uint8_t num_bytes = 32;
-  uint8_t buffer[num_bytes];
+  uint8_t buffer[COUNT_DATA_BYTES];
 
-  this->read_bytes_raw(buffer, num_bytes);
+  this->read_bytes_raw(buffer, COUNT_DATA_BYTES);
 
   // https://github.com/adafruit/Adafruit_PM25AQI
 
   // Check that start byte is correct!
-  if (buffer[0] != 0x42) {
+  if (buffer[0] != START_CHARACTER_1 || buffer[1] != START_CHARACTER_2) {
+    ESP_LOGW(TAG, "Start character mismatch: %02X %02X != %02X %02X", buffer[0], buffer[1], START_CHARACTER_1,
+             START_CHARACTER_2);
+    return false;
+  }
+
+  const uint16_t payload_length = (uint16_t(buffer[2]) << 8) | uint16_t(buffer[3]);
```

Comment:
Please use `encode_uint16` from `helpers.h` here:
```suggestion
  const uint16_t payload_length = encode_uint16(buffer[2], buffer[3]);
```
Please also be sure to add the include at the top!